components/usage/pkg/db/cost_center_test.go

```diff
@@ -58,26 +60,143 @@ func TestCostCenterManager_GetOrCreateCostCenter(t *testing.T) {
 
 func TestCostCenterManager_UpdateCostCenter(t *testing.T) {
 	conn := dbtest.ConnectForTests(t)
-	mnr := db.NewCostCenterManager(conn, db.DefaultSpendingLimit{
-		ForTeams: 0,
-		ForUsers: 500,
+	limits := db.DefaultSpendingLimit{
+		ForTeams:                  0,
+		ForUsers:                  500,
+		MinForIndividualsOnStripe: 1000,
+	}
+
+	t.Run("prevents updates to negative spending limit", func(t *testing.T) {
+		mnr := db.NewCostCenterManager(conn, limits)
+		userAttributionID := db.NewUserAttributionID(uuid.New().String())
+		teamAttributionID := db.NewTeamAttributionID(uuid.New().String())
+		cleanUp(t, conn, userAttributionID, teamAttributionID)
+
+		_, err := mnr.UpdateCostCenter(context.Background(), db.CostCenter{
+			ID:              userAttributionID,
+			BillingStrategy: db.CostCenter_Other,
+			SpendingLimit:   -1,
+		})
+		require.Error(t, err)
+		require.Equal(t, codes.InvalidArgument, status.Code(err))
+
+		_, err = mnr.UpdateCostCenter(context.Background(), db.CostCenter{
+			ID:              teamAttributionID,
+			BillingStrategy: db.CostCenter_Stripe,
+			SpendingLimit:   -1,
+		})
+		require.Error(t, err)
+		require.Equal(t, codes.InvalidArgument, status.Code(err))
 	})
-	team := db.NewTeamAttributionID(uuid.New().String())
-	cleanUp(t, conn, team)
-	teamCC, err := mnr.GetOrCreateCostCenter(context.Background(), team)
-	t.Cleanup(func() {
-		conn.Model(&db.CostCenter{}).Delete(teamCC)
+
+	t.Run("individual user on Other billing strategy cannot change spending limit of 500", func(t *testing.T) {
+		mnr := db.NewCostCenterManager(conn, limits)
+		userAttributionID := db.NewUserAttributionID(uuid.New().String())
+		cleanUp(t, conn, userAttributionID)
+
+		_, err := mnr.UpdateCostCenter(context.Background(), db.CostCenter{
+			ID:              userAttributionID,
+			BillingStrategy: db.CostCenter_Other,
+			SpendingLimit:   501,
+		})
+		require.Error(t, err)
+		require.Equal(t, codes.FailedPrecondition, status.Code(err))
+
 	})
-	require.NoError(t, err)
-	require.Equal(t, int32(0), teamCC.SpendingLimit)
 
-	teamCC.SpendingLimit = 2000
-	teamCC, err = mnr.UpdateCostCenter(context.Background(), teamCC)
-	require.NoError(t, err)
-	t.Cleanup(func() {
-		conn.Model(&db.CostCenter{}).Delete(teamCC)
+	t.Run("individual user upgrading to stripe gets default limit of 1000 and can increase, but not decrease", func(t *testing.T) {
```

Comment:
Nit: Reading the code, users don't "get" the default limit of 1000, but need to specify a limit >= 1000:

```suggestion
	t.Run("individual user upgrading to stripe can set a limit of 1000 or more, but not less than 1000", func(t *testing.T) {
```